src/main/daml/Daml/Finance/Claims/Rule.daml

```diff
@@ -0,0 +1,89 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Claims.Rule
+  ( Rule(..)
+  ) where
+
+import DA.Date (toDateUTC)
+import DA.Set (fromList)
+import DA.Text (sha256)
+import Daml.Finance.Claims.Util (isZero')
+import Daml.Finance.Claims.Util.Lifecycle (lifecycle, lifecycleClaims, netOnTag, splitPending, timeEvent)
+import Daml.Finance.Interface.Claims.Claim qualified as Claim (getAcquisitionTime)
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (R, exerciseInterfaceByKey, getKey)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (I, Evolve(..), View(..))
+import Daml.Finance.Interface.Lifecycle.Rule.NewVersion qualified as NewVersion (I, CreateNewVersion(..), GetLastEventTimestamp(..))
+import Daml.Finance.Interface.Types.Common (Parties, PartiesMap)
+import Daml.Finance.Interface.Util.Common (fetchInterfaceByKey)
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, flattenObservers)
+import Daml.Finance.Lifecycle.Effect (Effect(..))
+import Prelude hiding (exercise, key)
+
+-- | Rule to process a time update event.
```

Comment:
I would specify that this applies to instruments that are modelled using "on-the-fly" claims.